### PR TITLE
Add trusted-python-toolkit based on gramine base image.

### PR DESCRIPTION
**Pull request by Hanyu-Jin**
## Description

Build python toolkit image, which contains flask, numpy and pandas, based on gramine base image.

### 1. Why the change?

Add python dependencies required.

### 2. User API changes

Python examples are placed under /ppml/examples. You can run them on sgx through scripts under /ppml/start-scripts. 

### 3. Summary of the change 

Add two Dockerfiles and relevant scripts to build base image and custom image.
Add flask, numpy and pandas examples.

### 4. New dependencies

- [ ] New Python dependencies
       - flask
       - numpy
       - pandas

### CODE REVIEW

File: ppml/trusted-python-toolkit/custom-image/make-sgx.sh [hunk 1]
```diff
@@ -0,0 +1,10 @@
+#!/bin/bash
+
+echo SGX_MEM_SIZE:$SGX_MEM_SIZE
+echo SGX_LOG_LEVEL:$SGX_LOG_LEVEL
+echo ATTESTATION:$ATTESTATION
+if [[ "ENABLE_DCAP_ATTESTATION" == "false" ]]; then
+   sed -i 's/"dcap"/"none"/g' bash.manifest.template
+fi
+make SGX=1 DEBUG=1 THIS_DIR=/ppml/trusted-big-data-ml  SPARK_USER=root G_SGX_SIZE=$SGX_MEM_SIZE G_LOG_LEVEL=$SGX_LOG_LEVEL
```
Comment (glorysdj) [inline ppml/trusted-python-toolkit/custom-image/make-sgx.sh:9]: delete SPARK_USER=root

File: ppml/trusted-python-toolkit/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,34 @@
+ARG BASE_IMAGE_NAME
```
Comment (glorysdj) [inline ppml/trusted-python-toolkit/Dockerfile:1]: organize the code as below
ARG
ENV
ADD
RUN

File: ppml/trusted-python-toolkit/build-toolkit-base-image.sh [hunk 3]
```diff
@@ -0,0 +1,39 @@
+export HTTP_PROXY_HOST=http://child-prc.intel.com
```
Comment (glorysdj) [inline ppml/trusted-python-toolkit/build-toolkit-base-image.sh:1]: dont expose internal proxy

File: ppml/trusted-python-toolkit/build-toolkit-base-image.sh [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+    --build-arg BASE_IMAGE_TAG=$BASE_IMAGE_TAG \
+    -t ${TOOLKIT_IMAGE_NAME}:${TOOLKIT_IMAGE_TAG} -f ./Dockerfile ."
+
+No_Proxy_Modified="sudo docker build \
+    --build-arg BASE_IMAGE_NAME=$BASE_IMAGE_NAME \
+    --build-arg BASE_IMAGE_TAG=$BASE_IMAGE_TAG \
+    -t ${FL_IMAGE_NAME}:${FL_IMAGE_TAG} -f ./Dockerfile ."
+
+
+if [[ "$JDK_URL" == "http://your-http-url-to-download-jdk" ]] || [[ "$SPARK_JAR_REPO_URL" == "http://your_spark_jar_repo_url" ]]
```
Comment (glorysdj) [inline ppml/trusted-python-toolkit/build-toolkit-base-image.sh:27]: no need to install java

File: ppml/trusted-python-toolkit/entrypoint.sh [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+# Execute the container CMD under tini for better hygiene
+exec /usr/bin/tini -s -- "bash"
```
Comment (glorysdj) [inline ppml/trusted-python-toolkit/entrypoint.sh:20]: need to add attestation template methods
and a command can be signed and replaced at runtime

File: ppml/trusted-python-toolkit/examples/numpy/test-numpy.py [hunk 6]
```diff
@@ -0,0 +1,24 @@
+#!/usr/bin/env python3
```
Comment (glorysdj) [inline ppml/trusted-python-toolkit/examples/numpy/test-numpy.py:1]: rename to hello-numpy.py

File: ppml/trusted-python-toolkit/examples/pandas/test-pandas.py [hunk 7]
```diff
@@ -0,0 +1,17 @@
+#!/usr/bin/env python3
```
Comment (glorysdj) [inline ppml/trusted-python-toolkit/examples/pandas/test-pandas.py:1]: rename to hello-pandas.py

**Comment by glorysdj** [comment]
need to add some perf example for both numpy and pandas

File: ppml/trusted-python-toolkit/custom-image/make-sgx.sh (hunk 1 above)
Comment (Hanyu-Jin) [reply ppml/trusted-python-toolkit/custom-image/make-sgx.sh]: Fixed.

File: ppml/trusted-python-toolkit/Dockerfile (hunk 2 above)
Comment (Hanyu-Jin) [reply ppml/trusted-python-toolkit/Dockerfile:1]: Fixed.

File: ppml/trusted-python-toolkit/build-toolkit-base-image.sh (hunk 3 above)
Comment (Hanyu-Jin) [reply ppml/trusted-python-toolkit/build-toolkit-base-image.sh]: Fixed.

File: ppml/trusted-python-toolkit/build-toolkit-base-image.sh (hunk 4 above)
Comment (Hanyu-Jin) [reply ppml/trusted-python-toolkit/build-toolkit-base-image.sh]: Fixed.

File: ppml/trusted-python-toolkit/examples/numpy/test-numpy.py (hunk 6 above)
Comment (Hanyu-Jin) [reply ppml/trusted-python-toolkit/examples/numpy/test-numpy.py]: Fixed.

File: ppml/trusted-python-toolkit/examples/pandas/test-pandas.py (hunk 7 above)
Comment (Hanyu-Jin) [reply ppml/trusted-python-toolkit/examples/pandas/test-pandas.py]: Fixed.

File: ppml/trusted-python-toolkit/entrypoint.sh (hunk 5 above)
Comment (Hanyu-Jin) [reply ppml/trusted-python-toolkit/entrypoint.sh]: Fixed.

**Review by glorysdj — APPROVED** [review]
LGTM

**Pull request merged**